python/lib/dcm2bids_imaging_pipeline_lib/base_pipeline.py

```diff
@@ -413,12 +386,9 @@ def determine_phantom_data_site(self, string_with_site_acronym):
          :type string_with_site_acronym: str
         """
 
-        pscid = self.subject_id_dict["PSCID"]
-        visit_label = self.subject_id_dict["visitLabel"]
-
         # first check whether there is already a session in the database for the phantom scan
-        if pscid and visit_label:
-            return self.session_obj.get_session_center_info(pscid, visit_label)
+        if self.subject.psc_id and self.subject.visit_label:
```

Comment:
Same thing, this condition can never be false. This function is only called in `BasePipeline.create_session`, which is only called after `validate_subject_ids`, at which point the program would have exited long ago if there is no subject configuration provided.

This also means that the rest of this function is dead code.

Call of `BasePipeline.create_session`:
https://github.com/aces/Loris-MRI/blob/48231434b8ce9ed1f1f2091ced8673b3aabd2f2c/python/lib/dcm2bids_imaging_pipeline_lib/nifti_insertion_pipeline.py#L127

Reply:
it is possible that get_subject_id will return PSCID, CandiD and visit label but they are non existant in the database, in which case subject and or session should be created by the pipeline (if configuration to create candidate and session is turned on in the config module)

Reply:
Summary after our discussion: Although this can be called on non-validated subject configuration, the visit label is always used shortly after when creating the session (and must therefore exist and be valid). The PSCID could theoretically be `None` but that would be weird since it is known that is should be `"scanner"`.